R/assert.R

```diff
@@ -94,6 +97,15 @@ vec_is <- function(x, ptype = NULL, size = NULL) {
   TRUE
 }
 
+# Bare prototypes act as partial types. Only the SEXPTYPE is checked.
+is_same_type <- function(x, y) {
```

Comment:
I think this probably shouldn't have symmetric argument names, but instead be `x, ptype`, because this function will eventually check if `ptype` is a partial type, and do something special in that case.